### ISSUE TITLE
Add configuration option(s) to allow highlighting of huge files

### ISSUE THREAD

**Issue by mechatroner** [Microsoft/vscode#44986]
Hello!
I am the author of "Rainbow CSV" extension, and users of the extension have a problem because VS Code doesn't allow highlighting of big files: https://github.com/mechatroner/vscode_rainbow_csv/issues/3

Currently file size limits are hardcoded into the source ( MODEL_TOKENIZATION_LIMIT, MANY_MANY_LINES) and I suggest to either move them to VSCode user config or to add a boolean config option like "allowHugeFilesSyntax" which would be false by default, and if it is set to true, VSCode would ignore the hardcoded constants.

**Comment by alexandrudima** [comment]
Fixed via PR #44988

**Comment by isidorn** [comment]
I verified these limits work as expected, however the configurations are a bit too cryptic imho.
I would consider renaming them to have a word "limit" or "threshold" at the end
"editor.largeFileSizeLimit": 20971520,
	"editor.largeFileLineCountLimit": 300000,

Maybe even remove the word large
"editor.fileSizeThreshold": 20971520,
	"editor.fileLineCountThreshold": 300000,

Another thing which I dislike is the user specifying what is the exact threshold for him. Could we have made this a bollean which means no limit, and would have some warning message that opening huge files might make vscode unresponsive.
